### PR TITLE
tests: fix remodel-kernel test when running on external devices

### PR DESCRIPTION
**Pull request by mvo5**
Save/restore the tracking channel of the kernel in the test to
fix a failure when this test runs on e.g. external devices that
have models where the tracking channel of the kernel does not
match the "$KERNEL_CHANNEL" we use when we build the images on
our own.

### CODE REVIEW

File: tests/core/remodel-kernel/task.yaml [hunk 1]
```diff
@@ -16,7 +16,9 @@ prepare: |
     . "$TESTSLIB"/systemd.sh
 
     # Save the revision of the pc-kernel snap.
-    readlink /snap/pc-kernel/current > original-revision.txt
+    readlink /snap/"$OLD_KERNEL"/current > original-revision.txt
+    # Save the original tracking channel
+    snap list "$OLD_KERNEL" | grep -v ^Name | tr -s ' ' | cut -f4 -d' ' > original-channel.txt
```
Comment (bboozzoo) [inline tests/core/remodel-kernel/task.yaml:21]:
```suggestion
    snap list "$OLD_KERNEL" | grep -v ^Name | awk '{print $4}' > original-channel.txt
```

**Review by bboozzoo — APPROVED** [review]
LGTM

File: tests/core/remodel-kernel/task.yaml (hunk 1 above)
Comment (pedronis) [reply tests/core/remodel-kernel/task.yaml:21]: wouldn't snap info and grep tracking:  be slightly simpler?  not sure

**Review by pedronis — APPROVED** [review]
thanks

File: tests/core/remodel-kernel/task.yaml (hunk 1 above)
Comment (mvo5) [reply tests/core/remodel-kernel/task.yaml]: Nice suggestions, tweaked it to use both which makes it much nicer.

**Pull request merged**